### DISCUSSION TITLE
No sound or file produced

### DISCUSSION THREAD

**Discussion by dimdra** [rtl-airband/RTLSDR-Airband#302] [Q&A] [answered]
I am a newbie trying to configure airband using a rtl-sdr dongle. Apologies if I have missed something.

I have set it up but do not get either any sound of the icecast stream (test.mp3 mountpoint is visible in icecast when the application is run)nor an output file is created. I am on a raspberry pi 3+ buster.

Here is the config file:

```
devices:
({
  type = "rtlsdr";
  index = 0;
  gain = 0.0;
  centerfreq = 126.4;
  channels: (
  {
    freq = 126.4;
    outputs:
    ({
      type = "icecast";
      server = "127.0.0.1";
      port = 8000;
      mountpoint = "test.mp3";
      username = "xxxxxx";
      password = "yyyyyy";
     },
     {
      type = "file";
      directory = "root";
      filename_template = "TEST";
     }

);
  }
);
});
```

I tried with multiple frequencies too but with the same result.

Am I doing something wrong or issues with my build? Thanks in advance for your help

**Comment by szpajder** [comment]
This is most probably due to the gain being set to 0. Valid range for this parameter on RTL dongles is between 0 and 49.6 with useful values between 30 and 40, so start somewhere in this range.

Moved to discussions, since it doesn't seem to be a bug.

**Reply by dimdra** [reply]
Thanks, I've tried different gain levels supported by the dongle without any luck.

**Reply by starvald** [reply]
Your center frequency is the same as your listening frequency so you are hitting the center spike.

**Reply by dimdra** [reply]
Any combination of gain and center freq produces the same result, no sound or file being output.

**Comment by szpajder** [comment]
So what is your current gain setting? Beause the config file that you've posted shows 0.0 and this value gives you the least chance of receiving anything.

Also the directory path for writing files is set to `"root"`, which is not an absolute path so it might work or not, depending on which directory are you in when starting the program. The way to go is to specify a full path here - you probably want it to be set to `"/root"`  instead. This should work if you start the program as root - otherwise the program won't be able to create any files in this directory. If this is the case, it will log an error messages telling you about it.

The next question is whether rtl_airband has actually connected to your Icecast server. If it's not listening on the address/port that you've set or if it's incorrectly configured, then the connection will fail. And again - logs are your friend, so go check them out (more info on where to look is in the [Logging](https://github.com/szpajder/RTLSDR-Airband/wiki/Running-the-program#logging) section in the Wiki).

**Reply by dimdra** [reply]
I've tried multiple gain settings:  0.0, 20, 24, 25.4, 36.4, 38.6, 40.2, 48.0, 49.6, currently set to 38.6

I changed the centerfreq to 127 

Yes I realised the mistake in the path and I corrected it. I am logged in as root and running the application as root. The application starts successfully  and throws no errors but just numbers under the frequency. However, still no file is being created

_**EDIT: I added `continuous = true;` and the file creation started. However, it just had the small burst of static on it**_

As far as connecting to icecast is concerned, yes the connection is being made as:

1. the mountpoint /test.mp3 is being created in icecast. 
2. Also http://<ip_addr>:8000/test.mp3 shows the player. On clicking the play button after a brief amount of time the stream "plays" with a burst of static for a fraction of second and then goes quiet
3. the logs also show successful stream creation

The one error that I see in the log is the following:

`systemd-udevd[1176]:`` Process '/usr/sbin/th-cmd --socket /var/run/thd.socket --passfd --udev' failed with exit code 1.
`

Not sure if this is of any relevance? I also tested the frequency using rtl_fm and output the sound _temporarily_ to  the 3.5mm jack and I could here the sound on the connected speaker

**Comment by szpajder** [comment]
The log message from systemd-udevd is irrelevant.

What's on the channel that you've configured? Is it transmitting continuously? If so, you might need to set the squelch level manually, since the auto squelch only works when there are periods of silence on the channel.

"throws no errors but just numbers under the frequency" - please be more specific. Or post a screenshot of these numbers.

**Comment by dimdra** [comment, marked as answer]
126.4 is a non-continuous frequency. Here is the output when i execute airband:

Gain set to 49.6
![image](https://user-images.githubusercontent.com/96349449/146785875-7cf6aa14-97c9-4ae5-9f53-c5f0028e60f6.png)



Gain set to 38.6
![image](https://user-images.githubusercontent.com/96349449/146786535-f88bc0ee-2670-4d22-be73-d5052f3ac16c.png)

I will also try with a continuous frequency now with adjusting the squelch manually

**Reply by szpajder** [reply]
Signal level is equal to the noise level which is why the squelch does not open and there is no sound. Nothing wrong here.

The noise level looks a little high. What kind of antenna are you using? Any preamplifiers in there?

Please take a recording of the I/Q data from the dongle. Stop rtl_airband and execute:

```
rtlsdr -d 0 -s 2560000 -g 38 -f 126400000 output.cu8
```

Keep it running for a few seconds, then stop it with ctrl+c and share the resulting output.cu8 file. Don't make the recording too long, as the file will grow quickly. If possible, monitor the 126.4 MHz channel with another radio at that time and try to record a piece with some transmission in this channel.

**Reply by dimdra** [reply]
I now have reception, though faint. Guess I need to tweak my antenna position. I will also try multiple frequencies now.
It somehow just started working, thank you for your help. I will mark the thread closed.

**Comment by starvald** [comment]
that doesn't look right to me, for me a rtlsdr with no antenna i get  -52 @ gain 49.6

what does rtl_test report?

**Comment by dimdra** [comment]
![image](https://user-images.githubusercontent.com/96349449/146795499-ae16ebb0-5b73-431b-9815-d73694d326d6.png)

No further output thereafter

**Comment by starvald** [comment]
that looks ok. hmm.
